File: leetcode/rust/p1119-remove-vowels-from-a-string.rs
// LeetCode 1119. Remove Vowels from a String.

use std::collections::HashSet;

struct Solution {}

impl Solution {
    pub fn remove_vowels(mut s: String) -> String {
        let vowels = HashSet::from(['a', 'e', 'i', 'o', 'u']);
        s.retain(|x| !vowels.contains(&x));
        return s;
    }
}

fn main() {
    struct TestCase {
        s: String,
        exp: String,
    }

    let test_cases = vec![
        TestCase {
            s: "leetcodeisacommunityforcoders".to_string(),
            exp: "ltcdscmmntyfrcdrs".to_string(),
        },
        TestCase {
            s: "aeiou".to_string(),
            exp: "".to_string(),
        },
        TestCase {
            s: "a".to_string(),
            exp: "".to_string(),
        },
        TestCase {
            s: "b".to_string(),
            exp: "b".to_string(),
        },
    ];
    for tc in test_cases {
        let act = Solution::remove_vowels(tc.s.clone());
        if tc.exp != act {
            println!("FAIL. s: {}, exp: {}, act: {}", tc.s, tc.exp, act);
        }
    }
}

File: leetcode/cpp/p1119-remove-vowels-from-a-string.cpp
// LeetCode 1119. Remove Vowels from a String.
#include <algorithm>
#include <iostream>
#include <string>

class Solution {
public:
    std::string removeVowels(const std::string& s)
    {
        std::string ret;
        std::copy_if(s.begin(), s.end(), std::back_inserter(ret),
            [](char c) { return c != 'a' && c != 'e' && c != 'i' && c != 'o' && c != 'u'; });
        return ret;
    }
};

int main()
{
    struct TestCase {
        std::string s;
        std::string exp;
    };
    const TestCase test_cases[] = {
        { "leetcodeisacommunityforcoders", "ltcdscmmntyfrcdrs" },
        { "aeiou", "" },
        { "a", "" },
        { "b", "b" },
    };
    Solution solution;
    for (const auto& tc : test_cases) {
        const auto act = solution.removeVowels(tc.s);
        if (tc.exp != act) {
            std::cerr << "FAIL, s: " << tc.s
                      << ", exp: " << tc.exp
                      << ", act: " << act
                      << "\n";
        }
    }
    return 0;
}